### PR TITLE
BatchNorm and backprop consolidation in optimizers

### PR DESCRIPTION
**Pull request by adb262**

### CODE REVIEW

File: src/normalizers/batch_norm.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        for i in range(len(batch_X)):
+            for j in range(len(batch_X[i])):
+                stdev_cumulator[j] += (batch_X[i][j] - means[j]) ** 2
+
+        stds = [math.sqrt(x) / len(batch_X) for x in stdev_cumulator]
+        centered_x = [[(x[j] - means[j]) for j in range(len(x))] for x in batch_X]
+        normalized_x = [[x[j] / (stds[j]**2 + 1e-5) for j in range(len(x))] for x in centered_x]
+
+        scaled_x = [[(x[j] * self.gamma[j]) + self.beta[j] for j in range(len(x))] for x in normalized_x]
+        self.cache = (normalized_x, centered_x, means, stds)
```
Comment (adb262) [inline src/normalizers/batch_norm.py:29]: Unfinished
